packages/components/Checkbox/src/Checkbox.types.ts

```diff
@@ -0,0 +1,65 @@
+import * as React from 'react';
+import { IPressableProps } from '@fluentui-react-native/pressable';
+import { IPressableState } from '@fluentui-react-native/interactive-hooks';
+import { ViewProps } from 'react-native';
+import { IRenderData } from '@uifabricshared/foundation-composable';
+import { ITextProps } from '@fluentui-react-native/text';
+import { ITextTokens, IForegroundColorTokens, IBackgroundColorTokens, IBorderTokens } from '@fluentui-react-native/tokens';
+import { IViewWin32Props } from '@office-iss/react-native-win32';
+
+export const checkboxName = 'Checkbox';
+
+export interface ICheckboxInfo extends IPressableState {
+  checked: boolean;
+
+  disabled: boolean;
+
+  // To allow overrides in .settings. 'start' || undefined = false and 'end' = true
+  boxSide: boolean;
+}
+
+export interface ICheckboxState {
+  info: ICheckboxInfo;
+}
+
+export interface ICheckboxProps extends IPressableProps {
```

Comment:
Would something like what Button has work?
_export interface IButtonProps extends Omit<IPressableProps, 'onPress'>_

If not, where would this shared base props interface live?

Reply:
@FalseLobster what props are concerning here?

Reply:
I think the only thing we actually want form IPressableProps is `disabled ?: boolean`.  So I feel like that should be extracted to a shared type or maybe just explicitly added to the props interface since it's only one thing.

renderStyle, onPress, onStateChange, and the children type all feel a little weird to me.

Reply:
I removed the IPressableProps, I have disabled?: boolean; in the props.